internal/services/appservice/helpers/web_app_slot_schema.go

```diff
@@ -0,0 +1,519 @@
+	FtpsState               string                  `tfschema:"ftps_state"`
+	HealthCheckPath         string                  `tfschema:"health_check_path"`
+	HealthCheckEvictionTime int                     `tfschema:"health_check_eviction_time_in_min"`
+	NumberOfWorkers         int                     `tfschema:"number_of_workers"`
+	ApplicationStack        []ApplicationStackLinux `tfschema:"application_stack"`
+	MinTlsVersion           string                  `tfschema:"minimum_tls_version"`
+	ScmMinTlsVersion        string                  `tfschema:"scm_minimum_tls_version"`
+	Cors                    []CorsSetting           `tfschema:"cors"`
+	DetailedErrorLogging    bool                    `tfschema:"detailed_error_logging"`
+	LinuxFxVersion          string                  `tfschema:"linux_fx_version"`
+	VnetRouteAllEnabled     bool                    `tfschema:"vnet_route_all_enabled"`
+	// SiteLimits []SiteLimitsSettings `tfschema:"site_limits"` // TODO - New block to (possibly) support? No way to configure this in the portal?
+}
+
+func SiteConfigSchemaLinuxWebAppSlot() *pluginsdk.Schema {
+	return &pluginsdk.Schema{
+		Type:     pluginsdk.TypeList,
+		Required: true,
+		MaxItems: 1,
+		Elem: &pluginsdk.Resource{
+			Schema: map[string]*pluginsdk.Schema{
+				"always_on": {
+					Type:     pluginsdk.TypeBool,
+					Optional: true,
+					Default:  true,
+				},
+
+				"api_management_api_id": {
+					Type:         pluginsdk.TypeString,
+					Optional:     true,
+					ValidateFunc: apimValidate.ApiID,
+				},
+
+				"api_definition_url": {
+					Type:         pluginsdk.TypeString,
+					Optional:     true,
+					ValidateFunc: validation.IsURLWithHTTPorHTTPS,
+				},
+
+				"app_command_line": {
+					Type:     pluginsdk.TypeString,
+					Optional: true,
+				},
+
+				"application_stack": linuxApplicationStackSchema(),
+
+				"auto_heal": {
+					Type:     pluginsdk.TypeBool,
+					Optional: true,
+					RequiredWith: []string{
+						"site_config.0.auto_heal_setting",
+					},
+				},
+
+				"auto_heal_setting": autoHealSettingSchemaLinux(),
+
+				"container_registry_use_managed_identity": {
+					Type:     pluginsdk.TypeBool,
+					Optional: true,
+					Default:  false,
+				},
+
+				"container_registry_managed_identity_client_id": {
+					Type:         pluginsdk.TypeString,
+					Optional:     true,
+					ValidateFunc: msiValidate.UserAssignedIdentityID,
+				},
+
+				"default_documents": {
+					Type:     pluginsdk.TypeList,
+					Optional: true,
+					Computed: true,
+					Elem: &pluginsdk.Schema{
+						Type: pluginsdk.TypeString,
+					},
+				},
+
+				"http2_enabled": {
+					Type:     pluginsdk.TypeBool,
+					Optional: true,
+					Default:  false,
+				},
+
+				"ip_restriction": IpRestrictionSchema(),
+
+				"scm_use_main_ip_restriction": {
+					Type:     pluginsdk.TypeBool,
+					Optional: true,
+					Default:  false,
+				},
+
+				"scm_ip_restriction": IpRestrictionSchema(),
```

Comment:
should these conflict?

Reply:
The restrictions list can be updated independently of the switch to use it, so there are possibly valid use cases of this?